src/coreclr/src/jit/flowgraph.cpp

```diff
@@ -21911,6 +21910,12 @@ void Compiler::fgDebugCheckFlagsHelper(GenTree* tree, unsigned treeFlags, unsign
         //
         unsigned flagsToCheck = ~GTF_GLOB_REF & ~GTF_ORDER_SIDEEFF;
 
+        if (tree->OperIs(GT_IND) && (tree->gtGetOp1()->gtFlags & GTF_ICON_STATIC_HDL))
+        {
+            // Static readonly fields can be invariant
+            flagsToCheck &= ~GTF_IND_INVARIANT;
+        }
+
```

Comment:
Most statics won't be invariant.

Instead of giving up all checking for all static handles, could we introduce a `GTF_ICON_INVARIANT` to mark those that are invariant, and check selectively?

Reply:
@AndyAyersMS Should we take this current change and open up a issue fir follow on

Reply:
I've replace it with `GTF_ICON_CONST_PTR` (`constant is a pointer to immutable data, (e.g. IAT_PPVALUE)`)